### PR TITLE
Fix anonymous file upload

### PR DESCRIPTION
**Pull request by AdrienClairembault**
The changes introduced in https://github.com/glpi-project/glpi/commit/edb815973a4359c05d9ad1ae85706b79b128487e broke file upload for formcreator on public forms.

These changes revert this behavior for the `fileupload.php` endpoint, which mean we only check the `valid_id` session's data, which is generated correctly by formcreator in the context of public forms.

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets |

### CODE REVIEW

**Review by btry — APPROVED** [review]

**Comment by AdrienClairembault** [comment]
Last commit is needed if the form contain a "description" fields that contains images.
Without it the image wont load and the session will be destroyed (thus preventing the form to be submitted as the session CSRF data will be wiped).

image

File: front/document.send.php [hunk 1]
```diff
@@ -38,7 +38,8 @@
 include('../inc/includes.php');
 
 if (!$CFG_GLPI["use_public_faq"]) {
-    Session::checkLoginUser();
+    // Might be displaying image for anonymous users on formcreator
+    Session::checkValidSessionId(true);
```
Comment (cedric-anne) [inline front/document.send.php:42]: I probably miss something here.

`front/document.send.php` endpoint can be called using 2 distinct params.
 - Using the `file` param, only dumps, user pictures and inventory files can be served. Are you using this case ?
 - Using the `docid` param, `Document::canViewFile()` method will be used to validate that user has right to see the file. If there is no rights in session, will this method be able to return `true` ?

**Review by cedric-anne — CHANGES_REQUESTED** [review]

File: front/document.send.php (hunk 1 above)
Comment (AdrienClairembault) [reply front/document.send.php:42]: Full URL: `document.send.php?docid=11&itemtype=PluginFormcreatorQuestion&items_id=7`

I guess it falls under this condition:

image

Which leads to `PluginFormcreatorForm ::canView()`, which is always true:

image

**Review by cedric-anne — APPROVED** [review]

**Review by orthagh — CHANGES_REQUESTED** [review]
I'm not in favor of this pr atm.
Don't merge it, this requires discussions

**Comment by AdrienClairembault** [comment]
I'm thinking about other potentials solutions.

The cleanest way to do it would be through specific endpoints created by Formcreator that explicitly allow anonymous users.
However it isn't very practical as it would be hard to maintain (at least 4 endpoints would need to be maintained for things that should realistically be handled by GLPI core, not a plugin).

Another option would be to allow plugins to add a specific session flag to bypass the new check.
For example, setting `$_SESSION['do_not_check_profiles] = true;`
It feels dirty but it would be: 
* Extremely simple
* Let each plugins implement it anywhere they need it (no need to update core code once it's done)
* Do not worsen GLPI security (core)
* If done rights in plugins, it should not worsen their security either as it should be done only on pages that would be accessible without an account (thus not checking for a profile make sense and should not be exploitable)

We could even go on step further and name the parameter `$_SESSION['do_not_check_session]`, which mean Formcreator wouldn't even need to simulate a session anymore by setting some values like `glpiid`.

**Comment by trasher** [comment]
Will not be merged, as discussed

**Comment by trasher** [comment]
This is a plugin issue, and should be fixed in plugin. GLPI Core does not allow anonymous uploads anywhere.

This may be discussed, yes, but this is not the right place :)

**Comment by trasher** [comment]
(we've talked about that in our weekly meeting)

**Comment by cedric-anne** [comment]
It seems that people are tempted to apply this patch to fix `formcreator` anonymous form attachment functionality.
Anonymous uploads cannot be safe in GLPI context, so any patch that would fix this feature is a security flaw by nature.

**Pull request closed without merging**